packages/tailwindcss-language-service/src/completionProvider.ts

```diff
@@ -499,8 +499,8 @@ async function provideCustomClassNameCompletions(
   const positionOffset = document.offsetAt(position)
 
   const searchRange: Range = {
-    start: document.positionAt(Math.max(0, positionOffset - 1000)),
-    end: document.positionAt(positionOffset + 1000),
+    start: document.positionAt(0),
+    end: document.positionAt(positionOffset + 2000),
```

Comment:
Thanks for this PR @bradlc, it makes the situation with https://github.com/tailwindlabs/tailwindcss-intellisense/issues/837 much better.

Do you think if it’s possible to expose `2000` as an extension option? In our codebase, we have several instances of `cva()` which are pretty heavy on comments, so they end up being more than 2K characters long. 

I guess that if we look for `cva(` + `)`, the regexp might still not work. Or is it necessary to rewrite it? What do you reckon?

Reply:
Yeah I'm open to adding it as a config option. Can you share an example from your project that wouldn't be caught within the new range?

Reply:
The project is closed sourced, so I can’t share the snippet, sorry. You can imagine [chadcn → button.tsx]()(https://github.com/shadcn-ui/ui/blob/3c9f7ca0e20638bf75833ef920f9b23f5d97bc71/apps/www/registry/new-york/ui/button.tsx#L7-L35) (which is 1000 chars long), but with
- a few more variants, including [`compoundVariants`](https://cva.style/docs/getting-started/variants#compound-variants)
- extra spacing between groups of lines
- comments explaining the difference between variants
- a few more class names with selectors (`hover:`, `sm:`, `data-[state=xyz]:`, `[&>svg]:` and so on)

It’s quite easy to go beyond 2K symbols with all these.